### PR TITLE
NO-JIRA: [e2e test framework] Add a flag to add an annotation to HostedCluster

### PR DESCRIPTION
**Pull request by orenc1**
**What this PR does / why we need it**:
This addition will allow us to insert a custom annotation needed for the e2e test suite for the HostedCluster.
In particular, specify the management platform type in case of kubevirt external infra testing.

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/e2e_test.go [hunk 1]
```diff
@@ -115,6 +115,7 @@ func TestMain(m *testing.M) {
 	flag.StringVar(&globalOpts.ManagementClusterNamespace, "e2e.management-cluster-namespace", "", "Namespace of the management cluster's HostedCluster (required to test request serving isolation)")
 	flag.StringVar(&globalOpts.ManagementClusterName, "e2e.management-cluster-name", "", "Name of the management cluster's HostedCluster (required to test request serving isolation)")
 	flag.BoolVar(&globalOpts.DisablePKIReconciliation, "e2e.disable-pki-reconciliation", false, "If set, TestUpgradeControlPlane will upgrade the control plane without reconciling the pki components")
+	flag.StringVar(&globalOpts.configurableClusterOptions.Annotation, "e2e.annotation", "", "An annotation to apply to the HostedCluster (key=value)")
```
Comment (davidvossel) [inline test/e2e/e2e_test.go:118]: I'd suggest making the annotations repeatable.  That way we can add more than one annotation at a time. That will probably help us out in the future.

you can achieve this using a custom var with the flags package. i tested this out. here's an example.

```
package main

import (
        "flag" 
        "fmt"
        "strings"
)

type stringMapVar map[string]string

func (s *stringMapVar) String() string {
        if *s == nil {
                return ""
        }
        return fmt.Sprintf("%v", *s)
}

func (s *stringMapVar) Set(value string) error {
        split := strings.Split(value, "=")
        if len(split) != 2 {
                return fmt.Errorf("invalid argument: %s", value)
        }
        if *s == nil {
                *s = stringMapVar(map[string]string{})
        }
        map[string]string(*s)[split[0]] = split[1]
        return nil
}

var annotations stringMapVar

func main() {

        flag.Var(&annotations, "annotations", "")
        flag.Parse()

        if annotations != nil {
                fmt.Printf("%s\n", annotations.String())
        }
}

```

Comment (orenc1) [reply test/e2e/e2e_test.go]: done, thanks.
`e2e.annotations` can be specified multiple times and the annotations are aggregated.

**Review by davidvossel — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**